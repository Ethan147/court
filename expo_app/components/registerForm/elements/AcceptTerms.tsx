import React, { useState } from "react";
import {
  View,
  Text,
  TextStyle,
  ViewStyle,
  StyleProp,
  TouchableOpacity,
  Switch,
} from "react-native";
import { useNavigation, NavigationProp } from "@react-navigation/native";
import colors from "../../../utils/colors";

interface AcceptTermsProps {
  acceptTerms?: boolean;
  onPress?: () => void;
  passStyles?: {
    termsSwitchOuterView?: StyleProp<ViewStyle>;
    termsSwitchView?: StyleProp<ViewStyle>;
    termsTextView?: StyleProp<ViewStyle>;
    termsAgreeText?: StyleProp<TextStyle>;
  };
}

type TermsNavigationProp = NavigationProp<any, "TermsAndConditions">;

const AcceptTerms: React.FC<AcceptTermsProps> = ({
  acceptTerms = false,
  onPress,
  passStyles,
}) => {
  const styles = {
    termsSwitchOuterView: passStyles?.termsSwitchOuterView || {},
    termsSwitchView: passStyles?.termsSwitchView || {},
    termsTextView: passStyles?.termsTextView || {},
    termsAgreeText: passStyles?.termsAgreeText || {},
  };

  const navigation = useNavigation<TermsNavigationProp>();
  const [isAccepted, setIsAccepted] = useState(acceptTerms);

  const handlePress = () => {
    navigation.navigate("TermsAndConditions");
    if (onPress) {
      onPress();
    }
  };

  const handleSwitch = () => {
    setIsAccepted(!isAccepted);
  };

  return (
    <View style={styles?.termsSwitchOuterView}>
      <View style={styles?.termsSwitchView}>
        <Switch
          value={isAccepted}
          onValueChange={handleSwitch}
          ios_backgroundColor={colors.setting}
          trackColor={{ false: colors.settingSelect, true: colors.accent }}
          testID="switch"
        />
      </View>
      <View style={styles?.termsTextView}>
        <TouchableOpacity onPress={handlePress}>
          <Text style={styles?.termsAgreeText}>
            Agree to Terms and Conditions
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default AcceptTerms;
