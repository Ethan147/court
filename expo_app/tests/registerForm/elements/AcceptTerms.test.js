import { render, fireEvent } from "@testing-library/react-native";
import React from "react";

import AcceptTerms from "../../../components/registerForm/elements/AcceptTerms";

const mockNavigation = { navigate: jest.fn() };

jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useNavigation: () => mockNavigation,
}));

jest.mock("react-native-google-places-autocomplete", () => ({
  GooglePlacesAutocomplete: jest.fn(() => null),
}));

describe("AcceptTerms", () => {
  it("renders correctly", () => {
    const { getByText } = render(<AcceptTerms />);
    const linkElement = getByText(/Agree to Terms and Conditions/i);
    expect(linkElement).toBeTruthy();
  });

  it("calls the onPress prop and navigation when the terms and conditions link is pressed", () => {
    const onPress = jest.fn();

    // Set up the behavior of the mock function
    mockNavigation.navigate.mockReturnValue(null);

    const { getByText } = render(<AcceptTerms onPress={onPress} />);
    const termsLink = getByText(/Agree to Terms and Conditions/i);

    fireEvent.press(termsLink);

    expect(onPress).toHaveBeenCalled();
    expect(mockNavigation.navigate).toHaveBeenCalledWith("TermsAndConditions");
  });

  it("toggles the switch when clicked", () => {
    const { getByTestId } = render(<AcceptTerms />);
    const switchComponent = getByTestId("switch");

    fireEvent(switchComponent, "onValueChange", true);
    expect(switchComponent.props.value).toBe(true);

    fireEvent(switchComponent, "onValueChange", false);
    expect(switchComponent.props.value).toBe(false);

    fireEvent(switchComponent, "onValueChange", true);
    expect(switchComponent.props.value).toBe(true);
  });

  it("snapshot", () => {
    const tree = render(<AcceptTerms />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
