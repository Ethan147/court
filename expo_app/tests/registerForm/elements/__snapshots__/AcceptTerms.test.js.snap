// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcceptTerms snapshot 1`] = `
<View
  style={{}}
>
  <View
    style={{}}
  >
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#FFD700"
      style={
        [
          {
            "height": 31,
            "width": 51,
          },
          {
            "backgroundColor": "#F4EDE6",
            "borderRadius": 16,
          },
        ]
      }
      testID="switch"
      tintColor="#B3B3B3"
      value={false}
    />
  </View>
  <View
    style={{}}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={{}}
      >
        Agree to Terms and Conditions
      </Text>
    </View>
  </View>
</View>
`;
